probe/docker_process_mapper.go

```diff
@@ -50,63 +70,145 @@ var (
 )
 
 func (m *dockerMapper) update() {
-	pidTree, err := newPIDTreeStub(m.procRoot)
+	endpoint := "unix:///var/run/docker.sock"
+	client, err := newDockerClient(endpoint)
 	if err != nil {
 		log.Printf("docker mapper: %s", err)
 		return
 	}
 
-	endpoint := "unix:///var/run/docker.sock"
-	client, err := newDockerClient(endpoint)
-	if err != nil {
+	events := make(chan *docker.APIEvents)
+	if err := client.AddEventListener(events); err != nil {
 		log.Printf("docker mapper: %s", err)
 		return
 	}
+	defer func() {
+		if err := client.RemoveEventListener(events); err != nil {
+			log.Printf("docker mapper: %s", err)
+		}
+	}()
 
-	containers, err := client.ListContainers(docker.ListContainersOptions{All: true})
-	if err != nil {
+	if err := m.updateContainers(client); err != nil {
 		log.Printf("docker mapper: %s", err)
 		return
 	}
 
-	pmap := map[int]*docker.Container{}
-	for _, container := range containers {
-		info, err := client.InspectContainer(container.ID)
+	if err := m.updateImages(client); err != nil {
+		log.Printf("docker mapper: %s", err)
+		return
+	}
+
+	otherUpdates := time.Tick(m.interval)
+	for true {
```

Comment:
Yes we discussed this a while back, but I forgot to file a ticket.  This PR doesn't make the situation worse, so do you mind if we defer a Stop mechanism to another day?